pkg/webui/console/containers/applications-table/index.js

```diff
@@ -71,53 +64,161 @@ const tabs = [
     title: sharedMessages.allAdmin,
     name: ALL_TAB,
   },
+  { title: sharedMessages.deleted, name: DELETED_TAB },
 ]
 
-class ApplicationsTable extends Component {
-  constructor(props) {
-    super(props)
-
-    this.getApplicationsList = params => {
-      const { tab, query } = params
-
-      return getApplicationsList(params, ['name', 'description'], {
-        isSearch: tab === ALL_TAB || query.length > 0,
+const ApplicationsTable = props => {
+  const { isAdmin, restoreApplication, purgeApplication, ...rest } = props
+
+  const [tab, setTab] = React.useState(OWNED_TAB)
+  const isDeletedTab = tab === DELETED_TAB
+
+  const handleRestore = React.useCallback(
+    async id => {
+      try {
+        await restoreApplication(id)
+        toast({
+          title: id,
+          message: m.restoreSuccess,
+          type: toast.types.SUCCESS,
+        })
+      } catch (err) {
+        toast({
+          title: id,
+          message: m.restoreFail,
+          type: toast.types.ERROR,
+        })
+      }
+    },
+    [restoreApplication],
+  )
+
+  const handlePurge = React.useCallback(
+    async id => {
+      try {
+        await purgeApplication(id)
+        toast({
+          title: id,
+          message: m.purgeSuccess,
+          type: toast.types.SUCCESS,
+        })
+      } catch (err) {
+        toast({
+          title: id,
+          message: m.purgeFail,
+          type: toast.types.ERROR,
+        })
+      }
+    },
+    [purgeApplication],
+  )
+
+  const headers = React.useMemo(() => {
+    const baseHeaders = [
+      {
+        name: 'ids.application_id',
+        displayName: sharedMessages.id,
+        width: 25,
+        sortable: true,
+        sortKey: 'application_id',
+      },
+      {
+        name: 'name',
+        displayName: sharedMessages.name,
+        width: 25,
+        sortable: true,
+      },
+    ]
+
+    if (tab === DELETED_TAB) {
+      baseHeaders.push({
+        name: 'actions',
+        displayName: sharedMessages.actions,
+        width: 50,
+        getValue: row => ({
+          restore: handleRestore.bind(null, row.ids.application_id),
+          purge: handlePurge.bind(null, row.ids.application_id),
+        }),
+        render: handlers => (
+          <>
+            <Button message={sharedMessages.restore} secondary onClick={handlers.restore} />
+            <Button message={sharedMessages.purge} danger onClick={handlers.purge} />
```

Comment:
The purge action should be triggered through a `<DeleteModalButton />` that will warn about the implications of purging, the same way that purging does when doing it through the delete button directly.

The message displayed in the modal should however be reduced to just `m.modalPurgeWarning` and `m.purgeWarning`.